docs/api/history.md

```diff
@@ -0,0 +1,182 @@
+# History
+
+1. [Overview](#overview)
+2. [API](#api)
+
+**Note: This is a work in progress document and feature. History has been a
+private API for a very long time. We're still working through the best
+public interface for working with Microcosm's action history.
+
+## Overview
+
+All Microcosms have a history. This history keeps track of outstanding
+actions, working with a Microcosm to determine the next application
+state as actions move through different states.
+
+By default, Microcosm automatically disposes actions that are
+complete. However you can control how many actions Microcosm's history
+will hold on to by passing the `maxHistory` option:
+
+```javascript
+// Hold onto the last 10 actions
+let repo = new Microcosm({ maxHistory: 10 })
+
+// Or never forget anything
+let repo = new Microcosm({ maxHistory: Infinity })
+
+// Then push some actions
+repo.push(action)
+repo.push(action)
+repo.push(action)
+
+// Get actions out by using `repo.history.toArray()`
+let actions = repo.history.toArray() // [actionOne, actionTwo, actionThree]
+```
+
+### The data structure
+
+History stores actions as a tree, with a root and a head:
+
+```
+[root] - [one] - [two] - [three]
+```
+
+You can revert to a prior action by using `checkout(action)`:
+
+```javascript
+let repo = new Microcosm({ maxHistory: Infinity })
+
+let one = repo.push(action)
+let two = repo.push(action)
+
+repo.history.checkout(one)
+
+let three = repo.push(action)
+```
+
+In the example above, the following tree would be produced:
+
+```
+               |- [two]
+[root] - [one] +
+               |- [*three]
+```
+
+`three` is the active branch, marked with an asterisk. The active
+branch is used to determine state. Here, a Microcosm's state will be
+the result of dispatching `root`, `one`, and `three` to Domains.
+
+## API
+
+### checkout(action)
+
+Set the head of the tree to a target action. This has the effect
+of controlling time in a Microcosm's history:
+
+```javascript
+let repo = new Microcosm({ maxHistory: Infinity })
+
+let one = repo.push(action)
+let two = repo.push(action)
+
+repo.history.checkout(one)
+
+let three = repo.push(action)
+```
+
+The head of the history tree above is now three. Microcosm will
+calculate state by reconciling `root`, `one,` and `three`.
+
+
+### toggle([ ...actions ])
+
+Disable a group of actions:
```

Comment:
ah nice, i'd recommend documenting that it goes both ways. I have no idea what the use case for this would be so no opinion on the more targeted methods